packages/react-components/react-tooltip-v2-preview/library/src/components/Tooltip/useTooltip.tsx

```diff
@@ -0,0 +1,317 @@
+import { Escape } from '@fluentui/keyboard-keys';
+
+/**
+ * Create the state required to render Tooltip.
+ *
+ * The returned state can be modified with hooks such as useTooltipStyles_unstable,
+ * before being passed to renderTooltip_unstable.
+ *
+ * @param props - props from this instance of Tooltip
+ */
+export const useTooltip_unstable = (props: TooltipProps): TooltipState => {
+  'use no memo';
+
+  const context = useTooltipVisibility();
+  const isServerSideRender = useIsSSR();
+  const { targetDocument } = useFluent();
+  const [setDelayTimeout, clearDelayTimeout] = useTimeout();
+
+  const {
+    appearance = 'normal',
+    children,
+    content,
+    withArrow = false,
+    positioning = 'above',
+    onVisibleChange,
+    relationship,
+    showDelay = 250,
+    hideDelay = 250,
+    mountNode,
+  } = props;
+
+  const [visible, setVisibleInternal] = useControllableState({ state: props.visible, initialState: false });
+  const setVisible = React.useCallback(
+    (ev: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement> | undefined, data: OnVisibleChangeData) => {
+      clearDelayTimeout();
+      setVisibleInternal(oldVisible => {
+        if (data.visible !== oldVisible) {
+          onVisibleChange?.(ev, data);
+        }
+        return data.visible;
+      });
+    },
+    [clearDelayTimeout, setVisibleInternal, onVisibleChange],
+  );
+
+  const state: TooltipState = {
+    withArrow,
+    positioning,
+    showDelay,
+    hideDelay,
+    relationship,
+    visible,
+    shouldRenderTooltip: visible,
+    appearance,
+    mountNode,
+    // Slots
+    components: {
+      content: 'div',
+    },
+    content: slot.always(content, {
+      defaultProps: {
+        role: 'tooltip',
+        // @ts-expect-error
+        popover: 'manual',
+        onToggle: () => {
+          console.log('toggle');
+        },
+      },
+      elementType: 'div',
+    }),
+  };
+
+  const containerRef = React.useCallback((elem: HTMLDivElement) => {
+    return;
+  }, []);
+  const popoverRef = React.useRef<HTMLDivElement>(null);
+  state.content.ref = useMergedRefs(state.content.ref, containerRef, popoverRef);
+
+  const tooltipId = useId('tooltip-', state.content.id);
+
+  state.content.id = tooltipId;
+
+  const positioningOptions = {
+    arrowPadding: 2 * tooltipBorderRadius,
+    position: 'above' as const,
+    align: 'center' as const,
+    offset: 4,
+    ...resolvePositioningShorthand(positioning),
+  };
+
+  if (state.withArrow) {
+    positioningOptions.offset = mergeArrowOffset(positioningOptions.offset, arrowHeight);
+  }
+
+  const anchorProps = toPositionArea(positioningOptions.align, positioningOptions.position, positioningOptions.offset);
+
+  state.content.style = {
+    // @ts-expect-error
+    positionAnchor: `--${tooltipId}`,
+    ...anchorProps,
+    ...state.content.style,
+  };
+
+  // When this tooltip is visible, hide any other tooltips, and register it
```

Comment:
Do you think it's possible to enforce this with `auto`? https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/popover#auto

In other words, could we eliminate all this code to hide other tooltips?